lib/mssql.js

```diff
@@ -0,0 +1,215 @@
+import Ajv from "ajv";
+import JSONStream from "JSONStream";
+import {json} from "micro";
+import mssql from "mssql";
+import {notFound, failedCheck, badRequest} from "./errors.js";
+
+const TYPES = mssql.TYPES;
+
+const pools = new Map();
+
+// TODO: Confirm these are the right SQL verbs.
+const READ_ONLY = new Set(["SELECT", "SHOW DATABASES", "SHOW VIEW", "USAGE"]);
+
+const ajv = new Ajv();
+const validate = ajv.compile({
+  type: "object",
+  additionalProperties: false,
+  required: ["sql"],
+  properties: {
+    sql: {type: "string", minLength: 1},
+    params: {type: "array"},
+  },
+});
+
+// Since we want to work with more than one DB,
+// we will need to use a custom version of the MSSQL pool.
+// See: https://tediousjs.github.io/node-mssql/#connection-pools
+export const mssqlPool = {
+  get: (name, config) => {
+    if (!pools.has(name)) {
+      if (!config) {
+        throw new Error("Pool does not exist");
```

Comment:
A more helpful error message might be "Database configuration required", but I'm also confused by how we are calling this function?